browserstack_examples_specflowplus/BrowserStack/src/Hooks.cs

```diff
@@ -0,0 +1,154 @@
+﻿using BoDi;
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text;
+using TechTalk.SpecFlow;
+using BrowserStack.WebDriver.Core;
+using log4net;
+using NUnit.Framework;
+using OpenQA.Selenium;
+using System.Threading;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Threading;
+using BrowserStack.WebDriver.Core;
+using log4net;
+using NUnit.Framework;
+using OpenQA.Selenium;
+using OpenQA.Selenium.Chrome;
+
+namespace SpecflowBrowserStack.src.stepdefs
+{
+	[Binding]
+	public class Hooks
+	{
+		private static FeatureContext _featureContext;
+		private static ScenarioContext _scenarioContext;
+
+		public static ThreadLocal<IWebDriver> threadLocalDriver = new ThreadLocal<IWebDriver>();
+		//protected IWebDriver driver;
+		private static readonly ILog log = LogManager.GetLogger(typeof(Hooks));
+
+		public static DriverOptions GetWebDriverObject()
+		{
+
+			DriverFactory webDriverFactory = DriverFactory.GetInstance();
+			List<BrowserStack.WebDriver.Config.Platform> list;
+			list = webDriverFactory.GetPlatforms();
+			DriverOptions fixtureArgs = null;
+			foreach (BrowserStack.WebDriver.Config.Platform platform in list)
+			{
+				fixtureArgs = webDriverFactory.CreateRemoteWebCapabilities(platform);
+				log.Info(("Initialising driver with capabilities : {}", fixtureArgs));
+
+			}
+
+			return fixtureArgs;
+		}
+
+		public Hooks(FeatureContext featureContext, ScenarioContext scenarioContext)
+		{
+			_featureContext = featureContext;
+			_scenarioContext = scenarioContext;
+		}
+
+		[BeforeTestRun]
+		public static void InitializeReport()
+		{
+			//Initialize Extent report before test starts

```

Comment:
remove this specific comment about ExtentReport

also change the method name to something more generic.

e.g. BeforeTestRun